pdns/dnsdistdist/tcpiohandler.cc

```diff
@@ -246,19 +246,30 @@ class OpenSSLTLSConnection: public TLSConnection
     }
 
     if (!SSL_set_fd(d_conn, d_socket)) {
+      SSL_free(d_conn);
+      d_conn = nullptr;
       throw std::runtime_error("Error assigning socket");
     }
 
     int res = 0;
     do {
       res = SSL_accept(d_conn);
       if (res < 0) {
-        handleIORequest(res, timeout);
+        try {
+          handleIORequest(res, timeout);
+        }
+        catch(const std::exception&) {
```

Comment:
maybe this should catch `(...)` instead, to also get `PDNSExceptions`?

Reply:
Done, thanks!